packages/storage/amplify_storage_s3_dart/test/storage_s3_service/task/s3_download_task_test.dart

```diff
@@ -266,6 +271,7 @@ void main() {
         await downloadTask.pause();
```

Comment:
Why is `start` not awaited on the line above?

Reply:
`task.start` is an async function that creates async actions internally, e.g.

* prefix resolution
* Getting response from the `getObject` call


It's called when creating a download operation, and so is `unawaited`, as it should not block returning the created operation. The final state of `task.start` is determined by `await task.result.`


`task.pause` internally awaits the async action (if any) that is created by `task.start.` Using `unawaited` (task.start)  here replicates the actual flow in the implementation.

Reply:
Isn't it more common for a user to call `pause` sometime after `start` has completed?

Reply:
A user is not aware of the existence of `task.start`, even the task itself is hidden and wrapped by the operation.

```dart
final operation = Amplify.Storage.downloadData(...); // creates a _task, and call _task.start()

// a developer can call operation.pause() any time after getting the operation reference 
operation.pause(); // calls _task.pause() and
                   // the pause logic starts executing only when the async
                   // actions created by _task.start() complete.
```

Reply:
Okay. Just wondering if waiting for the task to start makes a difference then? Ideally, it shouldn't right?

Reply:
Correct, it doesn't. The behaviors of pausing on the following

```dart
unawaited(downloadTask.start());
await downloadTask.pause();
// OR
await downloadTask.start();
await downloadTask.pause();
```
are identical.

Reply:
Cool - can we change it to the latter then? 😅 `unawaited` makes it seem like you're specifically trying to pause before start finishes.

Reply:
I'd like to keep it as is, as this is how it's being called when creating the operation :D will add a comment to clarify 👍